crates/dojo-core/src/world_factory.cairo

```diff
@@ -10,38 +10,89 @@ mod WorldFactory {
     use starknet::ContractAddress;
     use starknet::contract_address::ContractAddressIntoFelt252;
     use starknet::syscalls::deploy_syscall;
+    use starknet::get_caller_address;
 
     use dojo_core::interfaces::IWorldDispatcher;
     use dojo_core::interfaces::IWorldDispatcherTrait;
     use dojo_core::string::ShortString;
+    use dojo_core::auth::systems::Route;
 
     struct Storage {
         world_class_hash: ClassHash,
         executor_address: ContractAddress,
+        auth_components: LegacyMap::<usize, ClassHash>,
+        auth_systems: LegacyMap::<usize, ClassHash>,
+        auth_components_len: usize,
+        auth_systems_len: usize,
```

Comment:
Here the default authentication components and systems are stored via mapping and index. Iterating over the index and mapping will get an array of the class hashes